src/main/java/com/webservices/company/repository/CompanyRepository.java

```diff
@@ -55,15 +56,18 @@ public Company get(Long id) {
 
     @Override
     public Company update(Long id, Company updatedCompany) {
-        // Preparing the update statement
         PreparedStatementCreator preparedStatementCreator = (Connection connection) -> {
             PreparedStatement preparedStatement = connection.prepareStatement("UPDATE public.companies\n" +
                     "\tSET name=?, address=?, number_of_employees=?, \"dateFound\"=?, type_of_business=?\n" +
                     "\tWHERE id = ?");
             preparedStatement.setString(1, updatedCompany.getName());
             preparedStatement.setString(2, updatedCompany.getAddress());
             preparedStatement.setInt(3, updatedCompany.getNumberOfEmployees());
-            preparedStatement.setTimestamp(4, Timestamp.from(updatedCompany.getDateFound()));
+            if (updatedCompany.getDateFound() == null) {
+                preparedStatement.setTimestamp(4, Timestamp.from(op.orElse(instant)));
```

Comment:
this looks strange. If you want to set a date to some default field, that you dont need to add 
Instant instant = Instant.parse("2021-02-09T11:19:42.12Z");
    Optional<Instant> op = Optional.of(instant);
Instead you can create constants (final static) and set the value
`private static final Instant DEFAULT_DATE_FOUND = Instant.parse("2021-02-09T11:19:42.12Z")`
If you dont need it, than let it be null in case updatedCompany does not have found date